paystack/src/main/java/co/paystack/android/ui/AuthActivity.java

```diff
@@ -73,7 +73,12 @@ public void handleResponse() {
     protected void setupWebview() {
         setContentView(R.layout.co_paystack_android____activity_auth);
 
-        findViewById(R.id.iv_close).setOnClickListener(v -> finish());
+        findViewById(R.id.iv_close).setOnClickListener(v -> {
+            synchronized (si) {
+                si.notify();
```

Comment:
Should you not use `notifyAll()` here. Sonarcloud has identified this as a problem in its analysis of this PR.

Reply:
Fixed